src/gui/app.cpp

```diff
@@ -11,8 +11,148 @@
 #include "printinfo.h"
 #include "version.h"
 
+#include <vector>
+#include <functional>
+
+namespace {
+NeovimQt::MainWindow* s_lastActiveWindow{nullptr};
+std::vector<std::reference_wrapper<NeovimQt::MainWindow>> s_windows;
+int s_exitStatus{0};
+} // namespace
+
 namespace NeovimQt {
 
+struct ConnectorInitArgs {
+	enum class Type {
+		Embed,
+		Server,
+		Spawn,
+		Default
+	};
+
+	ConnectorInitArgs(const QCommandLineParser& parser, QStringList nvimArgs) noexcept
+		: type{ getConnectorType(parser) }
+		, timeout{ parser.value("timeout").toInt() }
+		, server{ parser.value("server") }
+		, nvim{ parser.value("nvim") }
+		, positionalArgs{ parser.positionalArguments() }
+		, neovimArgs{ std::move(nvimArgs) }
+	{
+
+	}
+
+	ConnectorInitArgs(
+		Type _type, int _timeout, QString _server, QString _nvim, QStringList nvimArgs) noexcept
+		: type{ _type }
+		, timeout{ _timeout }
+		, server{ std::move(_server) }
+		, nvim{ std::move(_nvim) }
+		, positionalArgs{}
+		, neovimArgs{std::move(nvimArgs)}
+	{
+	}
+
+	const Type type;
+	const int timeout;
+	const QString server;
+	const QString nvim;
+	const QStringList positionalArgs;
+	const QStringList neovimArgs;
+
+	Type getConnectorType(const QCommandLineParser& parser) noexcept
+	{
+		if (parser.isSet("server")) {
+			return ConnectorInitArgs::Type::Server;
+		}
+
+		if (parser.isSet("embed")) {
+			return ConnectorInitArgs::Type::Embed;
+		}
+
+		if (parser.isSet("spawn") && !parser.positionalArguments().isEmpty()) {
+			return ConnectorInitArgs::Type::Spawn;
+		}
+
+		return ConnectorInitArgs::Type::Default;
+	}
+};
+
+NeovimConnector* connectToRemoteNeovim(const ConnectorInitArgs& args) noexcept
```

Comment:
This function should be `static`, it is not used outside of this file (I think).

This applies to most new functions in this PR. Generally, functions not defined in a header should be `static`.

Reply:
You are right. Same reason as to how I forgot to move the variables to the unnamed namespace. Apparently I forgot these too. Sorry. I'll fix it.